app/assets/javascripts/discourse/app/components/load-more-accessible.gjs

```diff
@@ -0,0 +1,271 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { schedule } from "@ember/runloop";
+import { service } from "@ember/service";
+import LoadMore from "discourse/components/load-more";
+import { i18n } from "discourse-i18n";
+
+/**
+ * @component LoadMoreAccessible
+ *
+ * Extends the standard LoadMore component with screen reader accessibility.
+ * Provides both intersection observer based loading and manual load buttons
+ * for users who navigate via headings or other assistive technologies.
+ *
+ * @param {Function} @action - Function to call when loading more content
+ * @param {"above"|"below"} @direction - Direction of loading (affects button text and aria labels)
+ * @param {boolean} @enabled - Whether loading is enabled (default: true)
+ * @param {boolean} @canLoadMore - Whether more content is available to load
+ * @param {string} @loadingText - Optional custom loading text
+ */
+export default class LoadMoreAccessible extends Component {
+  @service capabilities;
+
+  @tracked isLoading = false;
+
+  get direction() {
+    return this.args.direction || "below";
+  }
+
+  get isLoadingAbove() {
+    return this.direction === "above";
+  }
+
+  get canLoadMore() {
+    return this.args.canLoadMore ?? true;
+  }
+
+  get enabled() {
+    return this.args.enabled ?? true;
+  }
+
+  get buttonLabel() {
+    if (this.args.loadingText) {
+      return this.args.loadingText;
+    }
+
+    return this.isLoadingAbove
+      ? i18n("post.load_more_posts_above")
+      : i18n("post.load_more_posts_below");
+  }
+
+  get buttonAriaLabel() {
+    return this.isLoadingAbove
+      ? i18n("post.sr_load_more_posts_above")
+      : i18n("post.sr_load_more_posts_below");
+  }
+
+  @action
+  async handleIntersectionLoad() {
+    return this.#loadWithFocusManagement();
+  }
+
+  async #loadWithFocusManagement() {
+    if (!this.enabled || !this.canLoadMore || this.isLoading) {
+      return;
+    }
+
+    try {
+      this.isLoading = true;
+
+      // Announce loading has started
+      this.#announceLoading();
+
+      // Get existing post numbers and current focus context before loading
+      const existingPostNumbers = this.#getExistingPostNumbers();
```

Comment:
We can modify the PostStream component to pass this information to the component as an argument